### PR TITLE
New Memset module to manage DNS zone records

### PR DESCRIPTION
**Pull request by glitchcrab**
##### SUMMARY
This PR provides a module to manage DNS zone records hosted at [Memset](https://www.memset.com/apidocs/methods_dns.html#dns.zone_record_create) .

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
 * memset_zone_record

##### ANSIBLE VERSION
```
2.6
```

### CODE REVIEW

**Comment by duncan9562** [comment]
Added test records - working as expected!

**Comment by ghost** [comment]
Works fine for me.

**Comment by PebbleOfParagon** [comment]
Added and removed DNS records with this. Works exactly as expected

File: lib/ansible/modules/cloud/memset/memset_zone_record.py [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+notes:
+  - Zones can be thought of as a logical group of domains, all of which share the
+    same DNS records (i.e. they point to the same IP). An API key generated via the
+    Memset customer control panel is needed with the following minimum scope -
+    I(dns.zone_create), I(dns.zone_delete), I(dns.zone_list).
+description:
+    - Manage DNS records in a Memset account.
+options:
+    state:
+        required: true
```
Comment (resmo) [inline lib/ansible/modules/cloud/memset/memset_zone_record.py]: not a blocker but the convention is to have `present` as default, please also consider this for your module.

File: lib/ansible/modules/cloud/memset/memset_zone_record.py [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+    Matching records can be cleanly deleted without affecting other
+    resource types, so this is pretty simple to achieve.
+    '''
+    has_changed, has_failed = False, False
+    msg, memset_api = None, None
+
+    # if we have any matches, delete them.
+    if records:
+        for zone_record in records:
+            if args['check_mode']:
```
Comment (resmo) [inline lib/ansible/modules/cloud/memset/memset_zone_record.py:269]: not sure if this works, did you test check mode? shouldn't this be `module.check_mode`?

**Review by resmo — COMMENTED** [review]
Is there a way to have multiple a records? Is this supported by the api?

needs_info

File: lib/ansible/modules/cloud/memset/memset_zone_record.py (hunk 2 above)
Comment (glitchcrab) [reply lib/ansible/modules/cloud/memset/memset_zone_record.py:269]: I add it to the args [here](https://github.com/ansible/ansible/pull/40392/files/c6299ea97e6f28f4bc9fc5c6089f4fff088e2965#diff-f980ee4d7a8fcc19fa363523d47d03c7R349)

File: lib/ansible/modules/cloud/memset/memset_zone_record.py (hunk 1 above)
Comment (glitchcrab) [reply lib/ansible/modules/cloud/memset/memset_zone_record.py]: Sure, I can do that.

**Comment by glitchcrab** [comment]
@resmo unfortunately not, the API only supports a single record at a time so it's a case of a new task each time (when testing i've been using with_items).

File: lib/ansible/modules/cloud/memset/memset_zone_record.py (hunk 2 above)
Comment (resmo) [reply lib/ansible/modules/cloud/memset/memset_zone_record.py:269]: ok, missed that!

**Review by resmo — APPROVED** [review]
shipit

**Comment by resmo** [comment]
counting 4 shipits.

**Comment by resmo** [comment]
Thanks!

**Comment by James-MS** [comment]
Works for me!

**Pull request merged**